apps/wallet/src/ui/app/components/error-boundary/index.tsx

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ErrorBoundary as ReactErrorBoundary } from 'react-error-boundary';
+import { useLocation } from 'react-router-dom';
+
+import Alert from '_components/alert';
+
+import type { ReactNode } from 'react';
+import type { FallbackProps } from 'react-error-boundary';
+
+function Fallback({ error }: FallbackProps) {
```

Comment:
Do we want some way for the user to be able to reset the error boundary? Could be useful in the event the error is recoverable through re-rendering.

Reply:
Added the location change as the default way to recover. (Changing the location will reset the error-boundary's state) I was thinking that we could extend this component when we have a more specific case. Now I think navigating away should be enough (or refreshing). Actually I wanted to avoid a try again button because I don't think it makes sense for all the cases. Happy to update it though if you have something in mind.

Reply:
Yeah I think this is probably fine. We can think about a button to retry in the future if we see cases where it would be useful.